packages/zapp/console/src/components/Executions/ExecutionDetails/ExecutionNodeViews.tsx

```diff
@@ -169,18 +168,16 @@ export const ExecutionNodeViews: React.FC<ExecutionNodeViewsProps> = ({ executio
       </Tabs>
       <NodeExecutionDetailsContextProvider workflowId={workflowId}>
         <NodeExecutionsByIdContext.Provider value={nodeExecutionsById}>
-          {nodeExecutions.length > 0 ? (
-            <div className={styles.nodesContainer}>
-              {tabState.value === tabs.nodes.id && (
-                <div className={styles.filters}>
-                  <ExecutionFilters {...filterState} />
-                </div>
-              )}
-              <WaitForQuery errorComponent={DataError} query={nodeExecutionsQuery}>
-                {() => renderTab(tabState.value)}
-              </WaitForQuery>
-            </div>
-          ) : null}
+          <div className={styles.nodesContainer}>
+            {tabState.value === tabs.nodes.id && (
+              <div className={styles.filters}>
+                <ExecutionFilters {...filterState} />
+              </div>
+            )}
+            <WaitForQuery errorComponent={DataError} query={nodeExecutionsQuery}>
```

Comment:
let's simplify and add `loadingComponent` here as well, hence we can get rid of `loading` state lines 148-150